### PR TITLE
Spinner: Change spinner icon when `prefers-reduced-motion` is set

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- changes the spinner to an hourglass when `prefers-reduced-motion` is set
- i ended up having to fiddle with the `matchMedia` mocks as well since they weren't working correctly 🤷 
  - see e.g. https://github.com/facebook/create-react-app/issues/10126#issuecomment-735272763

**Why do we need this feature?**

- better a11y

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by ashharrison90** [comment]
ok something weird is going on with the `matchMedia` mock

but that is definitely not a friday afternoon job 😂 will come back to it on monday 🤓

File: packages/grafana-ui/src/components/Spinner/Spinner.tsx [hunk 1]
```diff
@@ -80,7 +82,17 @@ export const Spinner = ({
...
+      <Icon
+        className={cx(
+          {
+            'fa-spin': !prefersReducedMotion,
+          },
+          iconClassName
+        )}
+        name={iconName}
+        size={size}
+        aria-label="loading spinner"
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Spinner/Spinner.tsx:94]: While where here should we get rid of this aria-label? 🥺

**Review by joshhunt — APPROVED** [review]
Nice 👍

File: packages/grafana-ui/src/components/Spinner/Spinner.tsx (hunk 1 above)
Comment (ashharrison90) [reply packages/grafana-ui/src/components/Spinner/Spinner.tsx:94]: does it need replacing with anything or are we just straight removing it? 🤔

Comment (joshhunt) [reply packages/grafana-ui/src/components/Spinner/Spinner.tsx:94]: Hmm... after a brief look, about 75% use the spinner without a text label (and so would require a aria-label). Maybe lets just change it to 'loading' for now?

**Pull request merged**